src/python/grpcio/tests/unit/framework/interfaces/links/test_utilities.py

```diff
@@ -51,7 +51,7 @@ def _safe_for_log_ticket(ticket):
       possibly features values like the string "<payload of length 972321>" in
       place of the actual values of the given ticket.
   """
-  if isinstance(ticket.payload, (basestring,)):
+  if isinstance(ticket.payload, (str,)):
```

Comment:
You should use the [future](https://pypi.python.org/pypi/future) which adds compatibility between Python 2.7 and Python 3.

You should put on top of the file:
```
from builtins import str
```